stores/milvus-documentstore/milvus_documentstore/milvus.py

```diff
@@ -127,269 +132,307 @@ def __init__(
-            logger.warning("Calling DocumentStore.update_embeddings() on an empty index")
-            return
+        for x in self.get_all_documents(
+            index, filters, return_embedding, batch_size, headers
+        ):
+            yield x
 
-        logger.info("Updating embeddings for %s docs...", document_count)
+    def get_all_labels(
+        self,
+        index: Optional[str] = None,
+        filters: Optional[FilterType] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> List[Label]:
+        index = index or self.collection_name
 
-        result = self._query(
-            index=index,
-            vector_ids=None,
-            batch_size=batch_size,
-            filters=filters,
-            only_documents_without_embedding=not update_existing_embeddings,
+        filters = (
+            LogicalFilterClause.parse(filters).convert_to_milvus() if filters else None
         )
-        batched_documents = get_batches_from_generator(result, batch_size)
-        with tqdm(
-            total=document_count, disable=not self.progress_bar, position=0, unit=" docs", desc="Updating Embedding"
-        ) as progress_bar:
-            for document_batch in batched_documents:
-                self._delete_vector_ids_from_milvus(documents=document_batch, index=index)
-
-                embeddings = retriever.embed_documents(document_batch)
-                self._validate_embeddings_shape(
-                    embeddings=embeddings, num_documents=len(document_batch), embedding_dim=self.embedding_dim
-                )
+        res = self.client.query(
+            collection_name=index, filter_expression=filters, partition_keys=[2]
+        )
+        res_labels = []
+        for x in res:
+            x.pop(EMPTY_FIELD, None)
+            x.pop("embedding", None)
+            res_labels.append(Label.from_dict(x))
+        return res_labels
+
+    def get_document_count(
+        self,
+        filters: Optional[FilterType] = None,
+        index: Optional[str] = None,
+        only_documents_without_embedding: bool = False,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> int:
+        if headers:
+            raise NotImplementedError("MilvusDocumentStore does not support headers.")
+
+        if filters is not None:
+            filters = (
+                LogicalFilterClause.parse(filters).convert_to_milvus()
+                if filters
+                else None
+            )
 
-                if self.cosine:
-                    self.normalize_embedding(embeddings)
+        index = index or self.collection_name
+        if index not in self.client.list_collections():
+            self._create_collection(index)
+            return 0
 
-                mutation_result = self.collection.insert([embeddings.tolist()])
+        if only_documents_without_embedding:
+            only_documents_without_embedding = [1]
+        else:
+            only_documents_without_embedding = [0, 1]
```

Comment:
Not sure we need the `only_documents_without_embedding` parameter. Usually, the `get_document_count` method returns the count of all documents (with and without embeddings) and the `get_embedding_count` method returns the count of documents with embeddings.

Reply:
This is something done in the rest of the document stores so I did it aswell:
[here](https://github.com/deepset-ai/haystack/blob/6a5fbb71189f010a2c5f24866496c347b1510818/haystack/document_stores/weaviate.py#LL752C16-L752C16)
[here](https://github.com/deepset-ai/haystack/blob/6a5fbb71189f010a2c5f24866496c347b1510818/haystack/document_stores/pinecone.py#LL249C7-L249C7)